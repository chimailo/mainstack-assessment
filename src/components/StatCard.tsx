import { Info } from "lucide-react";

interface StatCardProps {
  label: string;
  amount: string;
}

export function StatCard({ label, amount }: StatCardProps) {
  return (
    <div className="flex items-start justify-between py-4">
      <div>
        <p className="text-sm text-muted-foreground mb-2 font-medium">
          {label}
        </p>
        <p className="text-2xl font-bold tracking-tight">{amount}</p>
      </div>
      <Info className="h-4 w-4 text-muted-foreground" />
    </div>
  );
}
