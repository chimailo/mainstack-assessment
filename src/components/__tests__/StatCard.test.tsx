import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { StatCard } from "../StatCard";

describe("StatCard", () => {
  it("renders and matches snapshot", () => {
    const { container } = render(
      <StatCard label="Total sales" amount="$12,345" />
    );

    expect(container).toMatchSnapshot();
  });
});
