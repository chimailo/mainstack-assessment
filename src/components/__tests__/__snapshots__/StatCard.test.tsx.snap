// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StatCard > renders and matches snapshot 1`] = `
<div>
  <div
    class="flex items-start justify-between py-4"
  >
    <div>
      <p
        class="text-sm text-muted-foreground mb-2 font-medium"
      >
        Total sales
      </p>
      <p
        class="text-2xl font-bold tracking-tight"
      >
        $12,345
      </p>
    </div>
    <svg
      aria-hidden="true"
      class="lucide lucide-info h-4 w-4 text-muted-foreground"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <path
        d="M12 16v-4"
      />
      <path
        d="M12 8h.01"
      />
    </svg>
  </div>
</div>
`;
